File: mycollab-web/src/main/java/com/esofthead/mycollab/module/crm/events/CaseEvent.java
package com.esofthead.mycollab.module.crm.events;

import com.esofthead.mycollab.vaadin.events.ApplicationEvent;

@SuppressWarnings("serial")
public class CaseEvent {
	
	public static class Save extends ApplicationEvent {
		public Save(Object source, Object data) {
			super(source, data);
		}
	}

	public static class Search extends ApplicationEvent {
		public Search(Object source, Object data) {
			super(source, data);
		}
	}

	public static class GotoList extends ApplicationEvent {
		public GotoList(Object source, Object data) {
			super(source, data);
		}
	}
	
	public static class GotoAdd extends ApplicationEvent {
		public GotoAdd(Object source, Object data) {
			super(source, data);
		}
	}
	
	public static class GotoRead extends ApplicationEvent {
		public GotoRead(Object source, Object data) {
			super(source, data);
		}
	}
	
	public static class GotoEdit extends ApplicationEvent {
		public GotoEdit(Object source, Object data) {
			super(source, data);
		}
	}
}

File: mycollab-web/src/main/java/com/mycollab/module/crm/event/CaseEvent.kt
package com.mycollab.module.crm.event

import com.mycollab.events.ApplicationEvent

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object CaseEvent {
    class GotoAdd(source: Any, data: Any?) : ApplicationEvent(source, data)

    class GotoEdit(source: Any, data: Any?) : ApplicationEvent(source, data)

    class GotoList(source: Any, data: Any?) : ApplicationEvent(source, data)

    class GotoRead(source: Any, data: Any?) : ApplicationEvent(source, data)

    class Save(source: Any, data: Any?) : ApplicationEvent(source, data)

    class Search(source: Any, data: Any?) : ApplicationEvent(source, data)
}